intel_extension_for_transformers/llm/runtime/graph/core/shared_memory_ccl.hpp

```diff
@@ -171,14 +171,23 @@ void reduce_2_fp32_buffers(int num_elements, void* rank_0, void* rank_1) {
   }
 }
 
+static void parallel_memcpy(void* to, void* from, size_t n_bytes) __attribute__((target("avx512bw")));
+static void parallel_memcpy(void* to, void* from, size_t n_bytes) {
+#pragma omp parallel for
+  for (int i = 0; i < n_bytes; i += VECTOR_LENGTH_IN_BYTES) {
+    auto val = _mm256_loadu_si256((__m256i*)((char*)from + i));
+    _mm256_storeu_si256((__m256i*)((char*)to + i), val);
```

Comment:
Question: what will ```n_bytes``` usually be? 4K? 4M?
 
Concerns
1. this will break support on avx2-only machines, like SRF
2. this may not be always faster per oneDNN's experience https://github.com/oneapi-src/oneDNN/blob/2e5eab8358be890a6bc2d3a386b7289aba377927/src/cpu/simple_concat.cpp#L125

I will suggest to learn from oneDNN, memcpy for small size and ```pragma omp simd``` for larger size

Reply:
any update here?